bindings/cs/MongoDB.Libmongocrypt/AwsKeyId.cs

```diff
@@ -78,24 +121,41 @@ void IInternalKmsKeyId.SetCredentials(ContextSafeHandle context, Status status)
 
             try
             {
-                IntPtr keyPointer = (IntPtr)Marshal.StringToHGlobalAnsi(CustomerMasterKey);
+                IntPtr customerMasterKeyPointer = (IntPtr)Marshal.StringToHGlobalAnsi(CustomerMasterKey);
                 try
                 {
                     // Let mongocrypt run strlen
                     context.Check(
                         status,
-                        Library.mongocrypt_ctx_setopt_masterkey_aws(context, regionPointer, -1, keyPointer, -1));
-                    ((IInternalKmsKeyId) this).SetAlternateKeyNames(context, status);
+                        Library.mongocrypt_ctx_setopt_masterkey_aws(context, regionPointer, -1, customerMasterKeyPointer, -1));
                 }
                 finally
                 {
-                    Marshal.FreeHGlobal(keyPointer);
+                    Marshal.FreeHGlobal(customerMasterKeyPointer);
                 }
             }
             finally
             {
                 Marshal.FreeHGlobal(regionPointer);
             }
+
+            if (Endpoint != null)
+            {
+                IntPtr endPointKeyPointer = (IntPtr)Marshal.StringToHGlobalAnsi(Endpoint);
+                try
+                {
+                    // Let mongocrypt run strlen
+                    context.Check(
+                        status,
+                        Library.mongocrypt_ctx_setopt_masterkey_aws_endpoint(context, endPointKeyPointer, -1));
+                }
+                finally
+                {
+                    Marshal.FreeHGlobal(endPointKeyPointer);
+                }
+            }
+
+            ((IInternalKmsKeyId)this).SetAlternateKeyNames(context, status);
```

Comment:
I assume moving this here (outside of the try/finally block it used to be in) is safe?